### PR TITLE
Refactor test cases for Login Component, Session Model and Conflict Model

### PR DESCRIPTION
**Pull request by NereusWB922**
### Summary:

Fixes part of #85 

#### Type of change:

- 🧪 Tests Update

### Changes Made:

- Refactor test cases for Login Component
- Remove test cases for Conflict model (removed model)
- Refactor test cases for Session model

### Proposed Commit Message:

```
Refactor test cases

The test cases for the Login Component and Session Model are outdated. 

Additionally, the Conflict Model has been removed. 

Let's refactor the test cases for the Login Component and Session Model 
and remove the test cases for the Conflict Model.

```

### CODE REVIEW

File: tests/app/auth/login/login.component.spec.ts [hunk 1]
```diff
@@ -36,10 +36,45 @@ describe('LoginComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should start login process', () => {
-    component.startLoginProcess();
+  it('should call startLoginProcess with false in startPublicOnlyLoginProcess', () => {
```
Comment (luminousleek) [inline tests/app/auth/login/login.component.spec.ts:39]: Spec description is a bit confusing, maybe can rename it to something like `startPublicOnlyLoginProcess should call startLoginProcess with false`. Similar comment for the test spec which calls `startIncludePrivateLoginProcess` in line 47

File: tests/app/auth/login/login.component.spec.ts [hunk 2]
```diff
@@ -36,10 +36,45 @@ describe('LoginComponent', () => {
...
+  it('should call authService.startOAuthProcess on startLoginProcess', () => {
+    const hasPrivateConsent = false;
+
+    component.startLoginProcess(hasPrivateConsent);
+
+    expect(authServiceSpy.startOAuthProcess).toHaveBeenCalledWith(hasPrivateConsent);
+    expect(loggingServiceSpy.info).toHaveBeenCalledWith('LoginComponent: Beginning login process');
+  });
+
+  it('should call authService.changeAuthState and errorHandlingService.handleError', () => {
```
Comment (luminousleek) [inline tests/app/auth/login/login.component.spec.ts:64]: Maybe can change the spec name to `should call error handling methods when error is thrown` or something - just to make clear that you're testing the error handling.

File: tests/app/core/models/session-model.spec.ts [hunk 3]
```diff
@@ -20,67 +20,67 @@ describe('Session Model', () => {
...
         .subscribe({
           next: () => fail(),
           error: (err) => expect(err).toEqual(new Error(NO_VALID_OPEN_PHASES))
         });
     });
 
-    it('should throw error on session data with undefined repo for open phase', () => {
-      of({ ...BUG_REPORTING_PHASE_SESSION_DATA, phaseBugReporting: undefined })
+    it('should throw error on session data with invalid repo', () => {
+      of({ ...VALID_SESSION_DATA, sessionRepo: [{ phase: Phase.issuesViewer, repo: undefined }] })
```
Comment (luminousleek) [inline tests/app/core/models/session-model.spec.ts:54]: This might be a good place to use spread syntax to override object properties but probably something like this
```
of({ sessionRepo: [{ ...ISSUES_VIEWER_SESSION_REPO, repo: undefined }] })
```
but I'll also be fine if you just do
```
of({ sessionRepo: [{ phase: Phase.issuesViewer, repo: undefined}] })
```

File: angular.json [hunk 4]
```diff
@@ -122,7 +122,7 @@
             "tsConfig": "src/tsconfig.spec.json",
             "karmaConfig": "tests/karma.conf.js",
             "scripts": [],
-            "styles": ["./node_modules/@angular/material/prebuilt-themes/indigo-pink.css", "src/styles.css", "src/markdown.css"],
+            "styles": ["./node_modules/@angular/material/prebuilt-themes/indigo-pink.css", "src/styles.css"],
```
Comment (luminousleek) [inline angular.json:125]: Can leave this change for a separate PR?

**Review by luminousleek — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a few minor changes to make things clearer.

**Comment by luminousleek** [comment]
Actually this PR could have been broken down further into 2 PRs: one for the Login component and one for the Session model. In fact I'll probably encourage you to break down the second part of your test refactoring PR to 4 separate PRs: one for each spec you're refactoring - makes our job as reviewers easier (since we can review 1 spec at a time without having to review all 4) and also means that reviews will be faster.

**Comment by NereusWB922** [comment]
> Actually this PR could have been broken down further into 2 PRs: one for the Login component and one for the Session model. In fact I'll probably encourage you to break down the second part of your test refactoring PR to 4 separate PRs: one for each spec you're refactoring - makes our job as reviewers easier (since we can review 1 spec at a time without having to review all 4) and also means that reviews will be faster.

ok got it. I will break down the second part later. Thanks for the review, will do the changes soon.

File: tests/app/auth/login/login.component.spec.ts [hunk 5]
```diff
@@ -36,10 +36,45 @@ describe('LoginComponent', () => {
...
+    spyOn(component, 'startLoginProcess');
 
-    expect(logger.info).toHaveBeenCalled();
-    expect(authService.startOAuthProcess).toHaveBeenCalled();
+    component.startPublicOnlyLoginProcess();
+
+    expect(component.startLoginProcess).toHaveBeenCalledWith(false);
+  });
+
+  it('should call startLoginProcess with true in startIncludePrivateLoginProcess', () => {
```
Comment (luminousleek) [inline tests/app/auth/login/login.component.spec.ts:47]: Same comment as the one I made in line 39 above.

File: tests/app/core/models/session-model.spec.ts [hunk 6]
```diff
@@ -20,67 +20,67 @@ describe('Session Model', () => {
...
-      of({ ...BUG_REPORTING_PHASE_SESSION_DATA, phaseBugReporting: undefined })
+    it('should throw error on session data with invalid repo', () => {
+      of({ sessionRepo: [{ phase: Phase.issuesViewer, repo: undefined }] })
         .pipe(assertSessionDataIntegrity())
         .subscribe({
           next: () => fail(),
           error: (err) => expect(err).toEqual(new Error(OPENED_PHASE_REPO_UNDEFINED))
         });
-      of({ ...BUG_REPORTING_PHASE_SESSION_DATA, phaseBugReporting: null })
+      of({ ...VALID_SESSION_DATA, sessionRepo: [{ phase: Phase.issuesViewer, repo: null }] })
```
Comment (luminousleek) [inline tests/app/core/models/session-model.spec.ts:60]: Same comment as the one I made for line 54 above, also applies to lines 66 and 72.

File: tests/constants/session.constants.ts [hunk 7]
```diff
@@ -1,25 +1,23 @@
...
-  ...BUG_REPORTING_PHASE_SESSION_DATA,
-  openPhases: []
+export const VALID_SESSION_DATA: SessionData = {
+  sessionRepo: [ISSUES_VIEWER_SESSION_REPO, ACTIVITY_DASHBOARD_SESSION_REPO]
 };
 
-export const MULTIPLE_OPEN_PHASES_SESSION_DATA: SessionData = {
-  ...BUG_REPORTING_PHASE_SESSION_DATA,
-  openPhases: [Phase.phaseBugReporting, Phase.phaseTeamResponse, Phase.phaseTesterResponse, Phase.phaseModeration]
+export const SESSION_DATA_WITH_INVALID_PHASE: SessionData = {
```
Comment (luminousleek) [inline tests/constants/session.constants.ts:21]: Is this constant being used in any of the tests?

**Review by luminousleek — CHANGES_REQUESTED** [review]
Thanks for making the changes! Still a few that you might have missed out.

File: tests/constants/session.constants.ts (hunk 7 above)
Comment (NereusWB922) [reply tests/constants/session.constants.ts:21]: No, will remove the constant.

**Review by luminousleek — APPROVED** [review]
LGTM, thanks!
